services/core-web/src/components/Forms/noticeOfWork/EditNoWContacts.js

```diff
@@ -83,11 +93,19 @@ const renderContacts = ({
                           contacts[index].party.name
                         )} as a contact on this Notice of Work?`}
                         onConfirm={() => {
-                          const updatedContact = contacts[index];
-                          updatedContact.state_modified = "delete";
+                          if (fields.get(index)) {
+                            // add state_modified and set to "delete" for backend
+                            contacts[index].state_modified = "delete";
+
+                            changeArray(FORM.EDIT_NOTICE_OF_WORK, "contacts", contacts);
 
-                          arrayPushReduxForms(FORM.EDIT_NOTICE_OF_WORK, "contacts", updatedContact);
-                          arrayRemoveReduxForms(FORM.EDIT_NOTICE_OF_WORK, "contacts", index);
+                            // move updated object, this will cause rerendering of the react component, setTimeout is required to bypass react optimization
+                            setTimeout(() => {
```

Comment:
This definitely a hack but it is the only way to rerender those FieldArray :fire:

Reply:
I never want to use field array again. Next time I'll create a custom component that deals with arrayPush, arrayChange, or change directly... seems like it will be easier. :/